lib/usd/translators/shading/mtlxFileTextureWriter.cpp

```diff
@@ -148,45 +150,67 @@ MtlxUsd_FileWriter::MtlxUsd_FileWriter(
     }
 
     // Now create a geompropvalue reader that the image shader will use.
-    TfToken primvarReaderName(
-        TfStringPrintf("%s_%s", _tokens->PrimvarReaderPrefix.GetText(), depNodeFn.name().asChar()));
-    const SdfPath  primvarReaderPath = nodegraphPath.AppendChild(primvarReaderName);
-    UsdShadeShader primvarReaderSchema = UsdShadeShader::Define(GetUsdStage(), primvarReaderPath);
-
-    primvarReaderSchema.CreateIdAttr(VtValue(TrMtlxTokens->ND_geompropvalue_vector2));
-
-    UsdShadeInput varnameInput
-        = primvarReaderSchema.CreateInput(TrMtlxTokens->geomprop, SdfValueTypeNames->String);
-
-    // We expose the primvar reader varname attribute to the material to allow
-    // easy specialization based on UV mappings to geometries:
-    SdfPath          materialPath = GetUsdPath().GetParentPath();
-    UsdShadeMaterial materialSchema(GetUsdStage()->GetPrimAtPath(materialPath));
-    while (!materialSchema && !materialPath.IsEmpty()) {
-        materialPath = materialPath.GetParentPath();
-        materialSchema = UsdShadeMaterial(GetUsdStage()->GetPrimAtPath(materialPath));
-    }
+    SdfPath primvarReaderPath = _GetPlace2DTexturePath(depNodeFn);
+
+    if (!GetUsdStage()->GetPrimAtPath(primvarReaderPath)) {
```

Comment:
Mostly indentation since we now detect shared place2dTexture nodes. Similar to what was done in usdFileTextureWriter.cpp in the first PR of the series.